### DISCUSSION TITLE
About three-phase to ground fault and line disconnection

### DISCUSSION THREAD

**Discussion by Crishawy** [CURENT/andes#249] [Q&A] [answered]
Dr Cui, 

In a 39-bus system, I added a three-phase to ground fault on bus7, followed by orderly line disconnection (t=0.11, line 11 and line 13), fault clearance (t=0.13) and reclosing (t=0.2) operations. Why do the voltages of nodes 6, 7, and 8 still change during the fault clearance stage after the lines (line11, line13) connected to bus7 are cut off? And, did I cut the lines the right way? 

Best regards!

Here are the volatages of bus 6, 7, and 8:
<img width="743" alt="voltage" src="https://user-images.githubusercontent.com/58496649/148912764-2fff5f3e-58da-4e7f-b1ff-de63fdec89fa.png">

Here are my main codes:

```tf = 1
no_output = False
tfault = 0.1  # fault start
t1 = 0.11  # line disconnection
tc = 0.13   # fault clearance
t2 = 0.2  # reclosing
tstep = 1 / 100  

ss = andes.run('ieee39_230_22.xlsx', setup=False)
ss.config.warn_limits = 0
ss.config.warn_abnormal = 0
fault_bus = ss.Line.bus1.v[12]
trip_line_1 = ss.Line.idx.v[10]
trip_line_2 = ss.Line.idx.v[12]
ss.add('Fault', {'bus': fault_bus, 'tf': tfault, 'tc': tc})
# cut lines
ss.add('Toggler', {'model': 'Line', 'dev': trip_line_1, 't': t1})
ss.add('Toggler', {'model': 'Line', 'dev': trip_line_2, 't': t1})
# reclosing
ss.add('Toggler', {'model': 'Line', 'dev': trip_line_1, 't': t2})
ss.add('Toggler', {'model': 'Line', 'dev': trip_line_2, 't': t2})
ss.setup()
ss.PFlow.run()
ss.TDS.init()
ss.TDS.config.tf = tf
ss.TDS.config.tstep = tstep
ss.TDS.run()
```

**Comment by cuihantao** [comment]
> did I cut the lines the right way?

It appears correct.

> Why do the voltages of nodes 6, 7, and 8 still change during the fault clearance stage after the lines (line11, line13) connected to bus7 are cut off?

Are you talking about the fast transient between 0.11 s and 0.13 s? It is the fast transients after changing cutting the lines.

Also, the way you plot it is not correct. Since the time stamps are not evenly spaced, you need to space them by the value, not the same spacing for consecutive samples. With that fixed, the transient will appear much faster.

`andes plot -a "v Bus" | xargs andes plot` should do the job if you have `xargs`. Otherwise, see notebook example 6.

**Reply by Crishawy** [reply]
Thanks, Cui! But I am still confused. I draw a new picture and the time points are marked. 
<img width="862" alt="1" src="https://user-images.githubusercontent.com/58496649/149454658-210a553f-f371-4d48-8f28-c113a50c3b0b.png">

I think after the operation of cuttiing lines, bus 6 and bus 8 have no connection to bus 7, so when bus 7 do fault clearance, the voltages of these two buses should not change. But the simulation results are just the opposite, which confuses me.

The following figure is a flowchart.
<img width="876" alt="2" src="https://user-images.githubusercontent.com/58496649/149456619-6dc940d2-8807-490d-bd15-5ea620058618.png">

**Reply by Crishawy** [reply]
Sorry, I loaded the first picture wrongly. This is the right one.
<img width="716" alt="重画图" src="https://user-images.githubusercontent.com/58496649/149456839-a6e13d70-9d61-4e54-ba75-17da0d18d5a1.png">

**Reply by cuihantao** [reply]
Is there any multi-line connection from 7 to 6 and 8 in the big circles that you didn't draw? Clearing the fault on bus 7 will reduce the reactive power output from 7. The effect on voltage will propagate throughout the network if ever connected.

**Reply by Crishawy** [reply]
Thank you for your patience！This is the complete graph. It seems that there are no other buses connected to bus 7. 
<img width="499" alt="微信图片_20220116150459" src="https://user-images.githubusercontent.com/58496649/149650701-f36265dd-579d-4fcc-b035-e6fd6b4a32c8.png">

**Comment by cuihantao** [comment]
Thanks for the drawing! 

It has to do with how fault clearance is implemented. By default, all bus voltages are restored to pre fault values upon fault clearance. This is to provide a good initial values for the voltages of post-fault equilibrium. See https://github.com/cuihantao/andes/blob/aa4567c2bfc2b80b5357316bd434717dd49dc37b/andes/models/timer.py#L178

In this case, clearing the fault on Bus 7 should not affect buses 6 or 8, as bus 7 is isolated. Can you try modifying the config file and change “restore” to 0 in section “[Fault]”?

**Reply by Crishawy** [reply]
Thanks, Cui!

After I set the value of restore to 0, there is no change in the voltages during the fault clearance process, which is in line with the theoretical results. But there are something wrong with the reclosing step.
![image](https://user-images.githubusercontent.com/58496649/149899551-456eb256-6ade-4e66-89e9-4d84ada4cf21.png)

**Comment by cuihantao** [comment]
Thanks again. I'm looking into this. It would be helpful if you can share your case. I'm currently starting from the original 39-bus system.

**Reply by Crishawy** [reply]
[ieee39_230_22.xlsx](https://github.com/cuihantao/andes/files/7894644/ieee39_230_22.xlsx)
This is my case.

**Comment by cuihantao** [comment, marked as answer]
After some initial investigation, I found two related issues:
1. As mentioned, all bus voltages are restored when the fault gets cleared. Bus 6 and 8 voltages are restored as well.
2. The power flow solutions can be attracted to the low-voltage or the high-voltage one, depending on the initial voltage values. 

When voltages are restored at 0.13 s, the system without bus 7 finds a high-voltage power flow. That's the reason the left figure has voltage changes on buses 6 and 8.

When voltages are not restored, the system does not have a good initial value to restore to the high-voltage solution after 0.2 s. All voltages remain low, as shown in your right figure.

The workaround is to clear the fault at 0.2 s, concurrent with the closure with lines. Voltages will be restored to the pre-fault one at 0.2 s correctly. I'm not sure of a permanent solution as `Togglers` are not aware of voltage value restoration yet. 

By the way, the voltage on Bus 7 when isolated has no mathematical meaning because there is no power flow to it.

**Reply by Crishawy** [reply]
Thank you very much, Cui! I have learned a lot.
